### ISSUE TITLE
Publishing with AOT for `linux-musl-x64` links an incorrect interpreter

### ISSUE THREAD

**Issue by giggio** [dotnet/runtime#93158]
### Description

If you publish an ASP .NET Core application with native AOT for rid `linux-musl-x64` it will link interpreter `/lib64/ld-linux-x86-64.so.2`, instead of musl's.

### Reproduction Steps

With .NET 8 RC1 (tested on Ubuntu 22.04):

1. `dotnet new web -o webmusl`
2. `cd webmusl`
3. `dotnet publish -r linux-musl-x64 -c Release -p:PublishAot=true`
4. `file ./bin/Release/net8.0/linux-musl-x64/publish/webmusl`

Output will be:

````
./bin/Release/net8.0/linux-musl-x64/publish/webmusl: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=85d93b90a1ebd5025459e71cfd78817c764b1393, for GNU/Linux 3.2.0, stripped
````

And application will run, even though Ubuntu is glibc based system.

### Expected behavior

Interpreter is set to musl's.

### Actual behavior

Interpreter is set to `linux-x64` rid.

### Configuration

.NET 8 RC1
Ubuntu 22.04.3 LTS
x64

**Comment by jkotas** [comment]
This is duplicate of https://github.com/dotnet/runtime/issues/92294

TLDR: The native AOT compilation depends on native toolchain on the system to be correctly configured for cross-compilation. It is not possible to reliably detect misconfigured native toolchain.
